pkg/build/package.go

```diff
@@ -653,6 +653,54 @@ func generatePkgConfigDeps(pc *PackageBuild, generated *config.Dependencies) err
 	return nil
 }
 
+// generatePythonDeps generates a python3~$VERSION dependency for packages which ship
+// Python modules.
+func generatePythonDeps(pc *PackageBuild, generated *config.Dependencies) error {
+	var pythonModuleVer string
+	pc.Logger.Printf("scanning for python modules...")
+
+	fsys := readlinkFS(pc.WorkspaceSubdir())
+	if err := fs.WalkDir(fsys, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		basename := filepath.Base(path)
+		if !strings.HasPrefix(basename, "python") {
+			return nil
+		}
+
+		// This probably shouldn't ever happen, but lets check to make sure.
+		if !d.IsDir() {
+			return nil
+		}
+
+		pythonModuleVer = basename[6:]
```

Comment:
What will this do if there happens to be a directory in the package named `python` for whatever reason? Would it create a runtime dep on `python~`? Is that allowed?

I think it would be great to have some test cases to act as runnable documentation about what this check expects to find and produce.

Reply:
It would set `pythonModuleVer` to `""`, which is the default anyway, thus no dependency would be added.